pkg/extension/sumologicextension/extension.go

```diff
@@ -101,39 +101,17 @@ func createHashKey(conf *Config) string {
 	return fmt.Sprintf("%s%s%s", conf.CollectorName, conf.Credentials.AccessID, conf.Credentials.AccessKey)
 }
 
+func (se *SumologicExtension) validateCredenials(
+	ctx context.Context,
+	creds api.OpenRegisterResponsePayload,
+) error {
+	return se.sendHeartbeat(ctx)
+}
+
 func (se *SumologicExtension) Start(ctx context.Context, host component.Host) error {
-	var colCreds CollectorCredentials
-	var err error
-	if se.credentialsStore.Check(se.hashKey) {
-		colCreds, err = se.credentialsStore.Get(se.hashKey)
-		if err != nil {
-			return err
-		}
-		se.collectorName = colCreds.CollectorName
-		if !se.conf.Clobber {
-			se.logger.Info("Found stored credentials, skipping registration",
-				zap.String("collector_name", colCreds.Credentials.CollectorName),
-			)
-		} else {
-			se.logger.Info(
-				"Locally stored credentials found, but clobber flag is set: " +
-					"re-registering the collector",
-			)
-			if colCreds, err = se.registerCollector(ctx, se.collectorName); err != nil {
-				return err
-			}
-			if err := se.credentialsStore.Store(se.hashKey, colCreds); err != nil {
-				se.logger.Error("Unable to store collector credentials", zap.Error(err))
-			}
-		}
-	} else {
-		se.logger.Info("Locally stored credentials not found, registering the collector")
-		if colCreds, err = se.registerCollector(ctx, se.collectorName); err != nil {
-			return err
-		}
-		if err := se.credentialsStore.Store(se.hashKey, colCreds); err != nil {
-			se.logger.Error("Unable to store collector credentials", zap.Error(err))
-		}
+	colCreds, registrationDone, err := se.getCredentials(ctx)
+	if err != nil {
+		return err
```

Comment:
I think we should add possibility to re-register in case of invalid cached credentials

Reply:
Created #120 to track this. Let's keep this PR as is (in terms of behavior) and introduce this in a separate PR when #120 is agreed on how it should behave.

Thanks!